Sync AliGPU over from O2 (fix End of Line character problem reported by Mikolaj)

davidrohr opened the pull request.

alibuild commented:
fd79cb9947366b8ef37b825d6f8b07e8e0399fc5: approval required: 1 of @dberzano (Dario Berzano), @ktf (Giulio Eulisse), @pzhristov (Peter Hristov), @davidrohr (David Rohr)

_Comment with `+1` to approve and allow automatic merging,or with `+test` to run tests only. **Please comment on the pull request: [click here](https://github.com/alisw/AliRoot/pull/961) and comment at the bottom of the page.**_

davidrohr commented:
+test

alibuild commented:
fd79cb9947366b8ef37b825d6f8b07e8e0399fc5: testing approved: will not be automatically merged; starting testing. If testing succeeds, merging will require further approval from 1 of @dberzano (Dario Berzano), @ktf (Giulio Eulisse), @pzhristov (Peter Hristov), @davidrohr (David Rohr)

alibuild commented:
Error while checking build/AliRoot/macos for fd79cb9947366b8ef37b825d6f8b07e8e0399fc5:
```
sw/BUILD/AliRoot-latest/log
 81/101 Test  #79: load_library_MUONcalign .................   Passed    6.62 sec
        Start  89: load_library_THepMCParser
 82/101 Test  #81: load_library_PHOSpi0Calib ...............   Passed    6.31 sec
        Start  90: load_library_HepMC
 83/101 Test  #83: load_library_AliStarLight ...............   Passed    6.28 sec
        Start  91: load_library_Photos
 84/101 Test  #84: load_library_AliGenFlat2Trk .............   Passed    6.04 sec
        Start  92: load_library_Tauola
 85/101 Test  #85: load_library_T0calib ....................   Passed    6.03 sec
        Start  93: load_library_TEvtGen
 86/101 Test  #86: load_library_T0shuttle ..................   Passed    6.30 sec
        Start  94: load_library_AliTPCSpaceChargeBase
 87/101 Test  #87: load_library_EvtGenExternal .............   Passed    5.43 sec
        Start  95: load_library_TPCcalib
 88/101 Test  #88: load_library_EvtGen .....................   Passed    5.25 sec
        Start  96: load_library_TPCmon
 89/101 Test  #89: load_library_THepMCParser ...............   Passed    5.18 sec
        Start  97: load_library_TPCupgrade
 90/101 Test  #90: load_library_HepMC ......................   Passed    5.10 sec
        Start  98: load_library_TRDcalib
 91/101 Test  #91: load_library_Photos .....................   Passed    5.29 sec
        Start  99: load_library_TRDgui
 92/101 Test  #92: load_library_Tauola .....................   Passed    5.34 sec
        Start 100: load_library_TRDqaAnalysis
 93/101 Test  #94: load_library_AliTPCSpaceChargeBase ......   Passed    5.36 sec
        Start 101: load_library_TTherminator
 94/101 Test  #93: load_library_TEvtGen ....................   Passed    6.32 sec
 95/101 Test  #96: load_library_TPCmon .....................   Passed    5.58 sec
 96/101 Test  #95: load_library_TPCcalib ...................   Passed    6.21 sec
 97/101 Test  #97: load_library_TPCupgrade .................   Passed    6.06 sec
 98/101 Test  #98: load_library_TRDcalib ...................   Passed    5.76 sec
 99/101 Test  #99: load_library_TRDgui .....................   Passed    4.39 sec
100/101 Test #100: load_library_TRDqaAnalysis ..............   Passed    4.69 sec
101/101 Test #101: load_library_TTherminator ...............   Passed    3.91 sec

91% tests passed, 9 tests failed out of 101

Total Test time (real) =  77.24 sec

The following tests FAILED:
	 35 - load_library_EveHLT (Failed)
	 45 - load_library_AliHLTCalo (Failed)
	 47 - load_library_AliHLTGlobal (Failed)
	 48 - load_library_AliHLTEMCAL (Failed)
	 53 - load_library_AliHLTPHOS (Failed)
	 57 - load_library_AliHLTTPCEVE (Failed)
	 58 - load_library_GPUTracking (Failed)
	 60 - load_library_AliHLTTPC (Failed)
	 66 - load_library_AliHLTEve (Failed)
```
Full log [here](https://ali-ci.cern.ch/repo/logs/alisw/AliRoot/961/fd79cb9947366b8ef37b825d6f8b07e8e0399fc5/build_AliRoot_macos/fullLog.txt).

alibuild commented:
55e57c33ef485135d577968830abc8a628d82ae5: approval required: 1 of @dberzano (Dario Berzano), @ktf (Giulio Eulisse), @pzhristov (Peter Hristov), @davidrohr (David Rohr)

_Comment with `+1` to approve and allow automatic merging,or with `+test` to run tests only. **Please comment on the pull request: [click here](https://github.com/alisw/AliRoot/pull/961) and comment at the bottom of the page.**_

davidrohr commented:
+test

alibuild commented:
55e57c33ef485135d577968830abc8a628d82ae5: testing approved: will not be automatically merged; starting testing. If testing succeeds, merging will require further approval from 1 of @dberzano (Dario Berzano), @ktf (Giulio Eulisse), @pzhristov (Peter Hristov), @davidrohr (David Rohr)

davidrohr commented:
@dberzano @ktf : Could you check what is the problem with the build/AliRoot/release CI, which reports "Cannot merge PR into test area"? This PR is already reabased onto alisw/AliRoot/master.

alibuild commented:
Error while checking build/AliRoot/release for 55e57c33ef485135d577968830abc8a628d82ae5:
```
sw/BUILD/AliRoot-latest/log
[ 84%] Generating libTEPEMGEN.rootmap
[ 84%] Built target libTEPEMGEN.rootmap
[ 84%] Building CXX object HLT/TPCLib/GPUTracking/CMakeFiles/GPUTracking.dir/Base/GPUProcessor.cxx.o
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliRoot/build_AliRoot_release/0/GPU/GPUTracking/Base/GPUReconstructionCPU.cxx:51:18: error: 'krnlRunRange' does not name a type
 static constexpr krnlRunRange GPUReconstructionCPU::krnlRunRangeNone;
                  ^~~~~~~~~~~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliRoot/build_AliRoot_release/0/GPU/GPUTracking/Base/GPUReconstructionCPU.cxx:52:18: error: 'krnlEvent' does not name a type; did you mean 'AliVEvent'?
 static constexpr krnlEvent GPUReconstructionCPU::krnlEventNone;
                  ^~~~~~~~~
                  AliVEvent
```
Full log [here](https://ali-ci.cern.ch/repo/logs/alisw/AliRoot/961/55e57c33ef485135d577968830abc8a628d82ae5/build_AliRoot_release/fullLog.txt).

alibuild commented:
Error while checking build/AliRoot/macos for 55e57c33ef485135d577968830abc8a628d82ae5:
```
sw/BUILD/AliRoot-latest/log
[ 95%] Building CXX object HLT/TPCLib/GPUTracking/CMakeFiles/GPUTracking.dir/Base/GPUSettings.cxx.o
[ 95%] Building CXX object HLT/TPCLib/GPUTracking/CMakeFiles/GPUTracking.dir/Base/GPUProcessor.cxx.o
[ 95%] Building CXX object HLT/TPCLib/GPUTracking/CMakeFiles/GPUTracking.dir/Base/GPUMemoryResource.cxx.o
/build/ci_checks/aliroot_checker_macos_next-root6_ci_0/sw/SOURCES/AliRoot/961/0/GPU/GPUTracking/Base/GPUReconstructionCPU.cxx:51:18: error: unknown type name 'krnlRunRange'; did you mean 'AliGPU::gpu::GPUReconstructionCPUBackend::krnlRunRange'?
static constexpr krnlRunRange GPUReconstructionCPU::krnlRunRangeNone;
                 ^~~~~~~~~~~~
                 AliGPU::gpu::GPUReconstructionCPUBackend::krnlRunRange
--
--
/build/ci_checks/aliroot_checker_macos_next-root6_ci_0/sw/SOURCES/AliRoot/961/0/GPU/GPUTracking/Base/GPUReconstruction.h:128:10: note: 'AliGPU::gpu::GPUReconstructionCPUBackend::krnlRunRange' declared here
  struct krnlRunRange {
         ^
/build/ci_checks/aliroot_checker_macos_next-root6_ci_0/sw/SOURCES/AliRoot/961/0/GPU/GPUTracking/Base/GPUReconstructionCPU.cxx:51:1: error: 'static' can only be specified inside the class definition
static constexpr krnlRunRange GPUReconstructionCPU::krnlRunRangeNone;
^~~~~~~
--
static constexpr krnlRunRange GPUReconstructionCPU::krnlRunRangeNone;
^~~~~~~
/build/ci_checks/aliroot_checker_macos_next-root6_ci_0/sw/SOURCES/AliRoot/961/0/GPU/GPUTracking/Base/GPUReconstructionCPU.cxx:52:18: error: unknown type name 'krnlEvent'; did you mean 'AliGPU::gpu::GPUReconstructionCPUBackend::krnlEvent'?
static constexpr krnlEvent GPUReconstructionCPU::krnlEventNone;
                 ^~~~~~~~~
                 AliGPU::gpu::GPUReconstructionCPUBackend::krnlEvent
--
--
/build/ci_checks/aliroot_checker_macos_next-root6_ci_0/sw/SOURCES/AliRoot/961/0/GPU/GPUTracking/Base/GPUReconstruction.h:136:10: note: 'AliGPU::gpu::GPUReconstructionCPUBackend::krnlEvent' declared here
  struct krnlEvent {
         ^
/build/ci_checks/aliroot_checker_macos_next-root6_ci_0/sw/SOURCES/AliRoot/961/0/GPU/GPUTracking/Base/GPUReconstructionCPU.cxx:52:1: error: 'static' can only be specified inside the class definition
static constexpr krnlEvent GPUReconstructionCPU::krnlEventNone;
^~~~~~~
[ 95%] Building CXX object HLT/TPCLib/GPUTracking/CMakeFiles/GPUTracking.dir/Base/GPUGeneralKernels.cxx.osw/BUILD/AliRoot-latest/log
[ 95%] Building CXX object HLT/TPCLib/GPUTracking/CMakeFiles/GPUTracking.dir/Base/GPUSettings.cxx.o
[ 95%] Building CXX object HLT/TPCLib/GPUTracking/CMakeFiles/GPUTracking.dir/Base/GPUProcessor.cxx.o
[ 95%] Building CXX object HLT/TPCLib/GPUTracking/CMakeFiles/GPUTracking.dir/Base/GPUMemoryResource.cxx.o
/build/ci_checks/aliroot_checker_macos_next-root6_ci_0/sw/SOURCES/AliRoot/961/0/GPU/GPUTracking/Base/GPUReconstructionCPU.cxx:51:18: error: unknown type name 'krnlRunRange'; did you mean 'AliGPU::gpu::GPUReconstructionCPUBackend::krnlRunRange'?
static constexpr krnlRunRange GPUReconstructionCPU::krnlRunRangeNone;
                 ^~~~~~~~~~~~
                 AliGPU::gpu::GPUReconstructionCPUBackend::krnlRunRange
--
--
/build/ci_checks/aliroot_checker_macos_next-root6_ci_0/sw/SOURCES/AliRoot/961/0/GPU/GPUTracking/Base/GPUReconstruction.h:128:10: note: 'AliGPU::gpu::GPUReconstructionCPUBackend::krnlRunRange' declared here
  struct krnlRunRange {
         ^
/build/ci_checks/aliroot_checker_macos_next-root6_ci_0/sw/SOURCES/AliRoot/961/0/GPU/GPUTracking/Base/GPUReconstructionCPU.cxx:51:1: error: 'static' can only be specified inside the class definition
static constexpr krnlRunRange GPUReconstructionCPU::krnlRunRangeNone;
^~~~~~~
--
static constexpr krnlRunRange GPUReconstructionCPU::krnlRunRangeNone;
^~~~~~~
```
Full log [here](https://ali-ci.cern.ch/repo/logs/alisw/AliRoot/961/55e57c33ef485135d577968830abc8a628d82ae5/build_AliRoot_macos/fullLog.txt).

alibuild commented:
11fea4b510d43c680c10302fd17ed245bb5ea245: approval required: 1 of @dberzano (Dario Berzano), @ktf (Giulio Eulisse), @pzhristov (Peter Hristov), @davidrohr (David Rohr)

_Comment with `+1` to approve and allow automatic merging,or with `+test` to run tests only. **Please comment on the pull request: [click here](https://github.com/alisw/AliRoot/pull/961) and comment at the bottom of the page.**_

davidrohr commented:
+test

alibuild commented:
11fea4b510d43c680c10302fd17ed245bb5ea245: testing approved: will not be automatically merged; starting testing. If testing succeeds, merging will require further approval from 1 of @dberzano (Dario Berzano), @ktf (Giulio Eulisse), @pzhristov (Peter Hristov), @davidrohr (David Rohr)

alibuild commented:
11fea4b510d43c680c10302fd17ed245bb5ea245: tests OK, approval required for merging: 1 of @dberzano (Dario Berzano), @ktf (Giulio Eulisse), @pzhristov (Peter Hristov), @davidrohr (David Rohr)

_Comment with `+1` to merge. **Please comment on the pull request: [click here](https://github.com/alisw/AliRoot/pull/961) and comment at the bottom of the page.**_

alibuild commented:
Error while checking build/AliRoot/release for 11fea4b510d43c680c10302fd17ed245bb5ea245:
```
sw/BUILD/AliRoot-guntest-latest/log
  *                                         *
  *  You are welcome to visit our Web site  *
  *          http://root.cern.ch            *
  *                                         *
  *******************************************

ROOT 5.34/30 (heads/v5-34-00-patches@v5-34-28-57-gec27989, Apr 30 2019, 12:48:00 on linuxx8664gcc)

CINT/ROOT C/C++ Interpreter version 5.18.00, July 2, 2010
Type ? for help. Commands must be C++ statements.
Enclose multiple statements between { }.

Processing rec.C...
W-AliRawReader::Create: No raw-reader created
I-AliReconstruction::InitRawReader: Reconstruction will run over digits
I-AliCDBManager::Init: AliEn classes enabled in Root. AliCDBGrid factory registered.
E-AliRunLoader::Open: Can not find run-Loader in file galice.root.
E-AliReconstruction::SetRunNumberFromData: No run loader found in file galice.root
W-AliQAManager::QAManager: You must specify kSIMMODE or kRECMODE or kQAMODE
E-AliReconstruction::Abort: SetRunNumberFromData failed! Aborting...
I-AliReconstruction::AbortProcess: SetRunNumberFromData failed! Aborting...
Real time 0:00:00, CP time 0.120
W-AliQAManager::QAManager: You must specify kSIMMODE or kRECMODE or kQAMODE
AliReconstruction
   InitCDB  R:0.0337s C:0.0300s (1 slices)
   InitRawReader  R:0.0109s C:0.0000s (1 slices)
   InitRun  R:0.1421s C:0.1200s (1 slices)
   Run  R:0.1421s C:0.1200s (1 slices)
  *******************************************
  *                                         *
  *        W E L C O M E  to  R O O T       *
  *                                         *
  *   Version   5.34/30     23 April 2015   *
  *                                         *
  *  You are welcome to visit our Web site  *
  *          http://root.cern.ch            *
  *                                         *
  *******************************************

ROOT 5.34/30 (heads/v5-34-00-patches@v5-34-28-57-gec27989, Apr 30 2019, 12:48:00 on linuxx8664gcc)

CINT/ROOT C/C++ Interpreter version 5.18.00, July 2, 2010
Type ? for help. Commands must be C++ statements.
Enclose multiple statements between { }.

Processing /mnt/mesos/sandbox/sandbox/sw/slc7_x86-64/AliRoot/build_AliRoot_release-1/STEER/macros/CheckESD.C...
E-AliRunLoader::Open: Can not find run-Loader in file galice.root.
E-CheckESD: getting run loader from file galice.root failed
(Bool_t)0
```
Full log [here](https://ali-ci.cern.ch/repo/logs/alisw/AliRoot/961/11fea4b510d43c680c10302fd17ed245bb5ea245/build_AliRoot_release/fullLog.txt).

alibuild commented:
62fbe2faf2a980e7f8b989bfa69de9acb6759b8f: approval required: 1 of @dberzano (Dario Berzano), @ktf (Giulio Eulisse), @pzhristov (Peter Hristov), @davidrohr (David Rohr)

_Comment with `+1` to approve and allow automatic merging,or with `+test` to run tests only. **Please comment on the pull request: [click here](https://github.com/alisw/AliRoot/pull/961) and comment at the bottom of the page.**_

davidrohr commented:
+test

alibuild commented:
62fbe2faf2a980e7f8b989bfa69de9acb6759b8f: testing approved: will not be automatically merged; starting testing. If testing succeeds, merging will require further approval from 1 of @dberzano (Dario Berzano), @ktf (Giulio Eulisse), @pzhristov (Peter Hristov), @davidrohr (David Rohr)

davidrohr commented:
@dberzano @ktf : Can you have a look why the PR cannot be merged into the tes area? In particular, the PR had passed the test already, and then became red due to "Cannot merge PR into test area". I force-pushed a new version in the meantime, but still get the same issue.

alibuild commented:
Error while checking build/AliRoot/release for 62fbe2faf2a980e7f8b989bfa69de9acb6759b8f:
```
sw/BUILD/AliRoot-latest/log
[ 69%] Building CXX object HLT/TPCLib/TPCFastTransformation/CMakeFiles/AliTPCFastTransformation.dir/IrregularSpline2D3D.cxx.o
In file included from /mnt/mesos/sandbox/sandbox/sw/SOURCES/AliRoot/build_AliRoot_release/0/GPU/TPCFastTransformation/IrregularSpline2D3D.cxx:22:0:
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliRoot/build_AliRoot_release/0/GPU/TPCFastTransformation/IrregularSpline2D3D.h: In member function 'void AliGPU::gpu::IrregularSpline2D3D::getSplineVec(const float*, float, float, float&, float&, float&) const':
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliRoot/build_AliRoot_release/0/GPU/TPCFastTransformation/IrregularSpline2D3D.h:355:18: error: 'conditional_t' in namespace 'std' does not name a template type
   using V = std::conditional_t<(Vc::float_v::size() >= 4),
                  ^~~~~~~~~~~~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliRoot/build_AliRoot_release/0/GPU/TPCFastTransformation/IrregularSpline2D3D.h:359:19: error: 'V' has not been declared
   float dataV[9 + V::size()];
                   ^
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliRoot/build_AliRoot_release/0/GPU/TPCFastTransformation/IrregularSpline2D3D.h:362:18: error: 'dataV' was not declared in this scope
   dataVvec.store(dataV, Vc::Unaligned);
                  ^~~~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliRoot/build_AliRoot_release/0/GPU/TPCFastTransformation/IrregularSpline2D3D.h:362:18: note: suggested alternative: 'dataV3'
   dataVvec.store(dataV, Vc::Unaligned);
                  ^~~~~
                  dataV3
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliRoot/build_AliRoot_release/0/GPU/TPCFastTransformation/IrregularSpline2D3D.h:364:37: error: 'V' has not been declared
   for (unsigned int i = 12; i < 9 + V::size(); i++) // fill not used part of the vector with 0
                                     ^
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliRoot/build_AliRoot_release/0/GPU/TPCFastTransformation/IrregularSpline2D3D.h:368:3: error: 'V' was not declared in this scope
   V dataU0vec(dataV), dataU1vec(dataV + 3), dataU2vec(dataV + 6), dataU3vec(dataV + 9);
   ^
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliRoot/build_AliRoot_release/0/GPU/TPCFastTransformation/IrregularSpline2D3D.h:368:3: note: suggested alternative: 'Vc'
   V dataU0vec(dataV), dataU1vec(dataV + 3), dataU2vec(dataV + 6), dataU3vec(dataV + 9);
   ^
   Vc
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliRoot/build_AliRoot_release/0/GPU/TPCFastTransformation/IrregularSpline2D3D.h:370:5: error: expected ';' before 'dataUVvec'
   V dataUVvec = gridU.getSpline(knotU, dataU0vec, dataU1vec, dataU2vec, dataU3vec, u);
     ^~~~~~~~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliRoot/build_AliRoot_release/0/GPU/TPCFastTransformation/IrregularSpline2D3D.h:372:7: error: 'dataUVvec' was not declared in this scope
   x = dataUVvec[0];
       ^~~~~~~~~
/mnt/mesos/sandbox/sandbox/sw/SOURCES/AliRoot/build_AliRoot_release/0/GPU/TPCFastTransformation/IrregularSpline2D3D.h:372:7: note: suggested alternative: 'dataVvec'
```
Full log [here](https://ali-ci.cern.ch/repo/logs/alisw/AliRoot/961/62fbe2faf2a980e7f8b989bfa69de9acb6759b8f/build_AliRoot_release/fullLog.txt).

davidrohr commented:
@ktf @dberzano : I removed the last commit and reverted back to the previous one, and alibuild directly merged it. Can you tell me why? Noone wrote +1.

> [11fea4b](https://github.com/alisw/AliRoot/commit/11fea4b510d43c680c10302fd17ed245bb5ea245): tests OK, approval required for merging: 1 of @dberzano (Dario Berzano), @ktf (Giulio Eulisse), @pzhristov (Peter Hristov), @davidrohr (David Rohr)

Is it because this had already been tested OK here? I think it should still wait for a +1!

alibuild commented:
Error while checking build/AliRoot/macos for 62fbe2faf2a980e7f8b989bfa69de9acb6759b8f:
```
sw/BUILD/AliRoot-latest/log
[ 72%] Building CXX object PHOS/PHOSbase/CMakeFiles/PHOSbase-object.dir/AliPHOSTriggerRawDigiProducer.cxx.o
[ 72%] Building CXX object PHOS/PHOSbase/CMakeFiles/PHOSbase-object.dir/AliPHOSTriggerRawReader.cxx.o
In file included from input_line_12:29:
/build/ci_checks/aliroot_checker_macos_next-root6_ci_0/sw/SOURCES/AliRoot/961/0/GPU/TPCFastTransformation/./IrregularSpline2D3D.h:355:18: error: no template named 'conditional_t' in namespace 'std'
  using V = std::conditional_t<(Vc::float_v::size() >= 4),
            ~~~~~^
--
  using V = std::conditional_t<(Vc::float_v::size() >= 4),
            ~~~~~^
/build/ci_checks/aliroot_checker_macos_next-root6_ci_0/sw/SOURCES/AliRoot/961/0/GPU/TPCFastTransformation/./IrregularSpline2D3D.h:359:19: error: use of undeclared identifier 'V'
  float dataV[9 + V::size()];
                  ^
--
  float dataV[9 + V::size()];
                  ^
/build/ci_checks/aliroot_checker_macos_next-root6_ci_0/sw/SOURCES/AliRoot/961/0/GPU/TPCFastTransformation/./IrregularSpline2D3D.h:364:37: error: use of undeclared identifier 'V'
  for (unsigned int i = 12; i < 9 + V::size(); i++) // fill not used part of the vector with 0
                                    ^
--
  for (unsigned int i = 12; i < 9 + V::size(); i++) // fill not used part of the vector with 0
                                    ^
/build/ci_checks/aliroot_checker_macos_next-root6_ci_0/sw/SOURCES/AliRoot/961/0/GPU/TPCFastTransformation/./IrregularSpline2D3D.h:368:3: error: unknown type name 'V'
  V dataU0vec(dataV), dataU1vec(dataV + 3), dataU2vec(dataV + 6), dataU3vec(dataV + 9);
  ^
--
  V dataU0vec(dataV), dataU1vec(dataV + 3), dataU2vec(dataV + 6), dataU3vec(dataV + 9);
  ^
/build/ci_checks/aliroot_checker_macos_next-root6_ci_0/sw/SOURCES/AliRoot/961/0/GPU/TPCFastTransformation/./IrregularSpline2D3D.h:368:33: error: expected ';' at end of declaration
  V dataU0vec(dataV), dataU1vec(dataV + 3), dataU2vec(dataV + 6), dataU3vec(dataV + 9);
                                ^
                                ;
--
--
  V dataU0vec(dataV), dataU1vec(dataV + 3), dataU2vec(dataV + 6), dataU3vec(dataV + 9);
                                ^
                                ;
/build/ci_checks/aliroot_checker_macos_next-root6_ci_0/sw/SOURCES/AliRoot/961/0/GPU/TPCFastTransformation/./IrregularSpline2D3D.h:370:3: error: unknown type name 'V'
  V dataUVvec = gridU.getSpline(knotU, dataU0vec, dataU1vec, dataU2vec, dataU3vec, u);
  ^
--
  V dataUVvec = gridU.getSpline(knotU, dataU0vec, dataU1vec, dataU2vec, dataU3vec, u);
  ^
/build/ci_checks/aliroot_checker_macos_next-root6_ci_0/sw/SOURCES/AliRoot/961/0/GPU/TPCFastTransformation/./IrregularSpline2D3D.h:370:51: error: use of undeclared identifier 'dataU1vec'
  V dataUVvec = gridU.getSpline(knotU, dataU0vec, dataU1vec, dataU2vec, dataU3vec, u);
                                                  ^
--
  V dataUVvec = gridU.getSpline(knotU, dataU0vec, dataU1vec, dataU2vec, dataU3vec, u);
                                                  ^
/build/ci_checks/aliroot_checker_macos_next-root6_ci_0/sw/SOURCES/AliRoot/961/0/GPU/TPCFastTransformation/./IrregularSpline2D3D.h:370:62: error: use of undeclared identifier 'dataU2vec'
```
Full log [here](https://ali-ci.cern.ch/repo/logs/alisw/AliRoot/961/62fbe2faf2a980e7f8b989bfa69de9acb6759b8f/build_AliRoot_macos/fullLog.txt).
